apps/showcase/src/components/training/code-editor-view/code-editor-view.component.ts

```diff
@@ -382,10 +405,33 @@ export class CodeEditorViewComponent implements OnDestroy {
     }
   }
 
+  public ngOnInit() {
+    const { cwd } = this.project();
+    const savedState = localStorage.getItem(cwd);
+    if (savedState) {
+      const modal = this.modalService.open(SaveCodeDialogComponent, { backdrop: 'static' });
+      void modal.result.then((positiveReply) => {
+        if (positiveReply) {
+          this.cwdTree$.pipe(take(1)).subscribe(async () => {
```

Comment:
```suggestion
          await firstValueFrom(this.cwdTree$);
```
?

Reply:
`ngOnInit` is not async, I could do it with a `.then` but it's the same no ?

Reply:
It's inside the `then((positiveReply) => ...)` already, just need to be`then(async (positiveReply) => ...)`.
But not problem to not do the change :D

Reply:
indeed I didn't remind that it was already in a `then`
I will do the change when I will rebase